File: estudos/kotlin/curso_kotlin_002_bytebank/src/main/kotlin/ContaPoupanca.kt
class ContaPoupanca(
    titular: String,
    numero: Int
) : Conta(
    titular,
    numero
) {
    override fun saca(valor: Double) {
        val valorComTaxa = valor + 0.1
        if(this.saldo >= valorComTaxa) {
            this.saldo -= valorComTaxa
        }
    }
}

File: Alura/estudos/javascript007/ContaPoupanca.js
import Conta from "./Conta.js";

export default class ContaPoupanca extends Conta {
    constructor(saldoInicial,cliente,agencia) {
        super(saldoInicial,cliente,agencia);
    };
};




























// sacar(valor) {
//     if(this._saldo >= valor) {
//         this._saldo -= valor;
//         return valor;
//     };
// };

// depositar(valor) {
//     if(valor < 100) {
//         return;
//     };
//     this._saldo += valor;
// };

// transferir(valor,conta) {
//     const valorSacado =  this.sacar(valor);
//     conta.depositar = valorSacado;
// };